test/type_prop/index_reduction.cpp

```diff
@@ -59,6 +59,50 @@ TEST(type_prop, index_reduction_invalid_rank)
     }
 }
 
+TEST(type_prop, argmin_invalid_zero_reduction_axis)
+{
+    auto a = make_shared<op::Parameter>(element::f32, Shape{2, 0});
+
+    try
+    {
+        auto argmin = make_shared<op::ArgMin>(a, 1, element::i32);
+        FAIL() << "ArgMin c-tor should throw for zero-legnth reduction axis";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), "reduction axis can not be empty");
+    }
+    catch (...)
+    {
+        FAIL() << "Deduced type check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, argmax_invalid_zero_reduction_axis)
+{
+    auto a = make_shared<op::Parameter>(element::f32, Shape{2, 0});
+
+    try
+    {
+        auto argmax = make_shared<op::ArgMax>(a, 1, element::i32);
+        FAIL() << "ArgMax c-tor should throw for zero-legnth reduction axis";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), "reduction axis can not be empty");
+    }
+    catch (...)
+    {
+        FAIL() << "Deduced type check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, sum_valid_zero_reduction_axis)
+{
+    auto a = make_shared<op::Parameter>(element::f32, Shape{2, 0});
+    auto sum = make_shared<op::Sum>(a, AxisSet{1});
```

Comment:
Might check the output shape here to make sure it is correct. Also, this doesn't really belong in `index_reduction.cpp`, because `Sum` is not an `IndexReduction` op.